### ISSUE TITLE
Big endian test failure (ruby-vips)

### ISSUE THREAD

**Issue by LocutusOfBorg** [libvips/libvips#1847]
Hello, this is not a big issue, because the test always failed there, so the Debian/Ubuntu autopkgtest infrastructure is not showing it as "regressed"

However I would like to point this possible issue in the code (or in the test code)
```
https://travis-ci.org/github/kleisauke/ruby-vips/jobs/733637121

  1) Vips::Image can load an image from a file

     Failure/Error: expect(x.avg).to eq(128)

       expected: 128

            got: 9.388699710976274e-44

       (compared using ==)

     # ./spec/image_spec.rb:27:in `block (2 levels) in <top (required)>'
```

**Comment by kleisauke** [comment]
fwiw, the pyvips test suite runs succefully on IBM Z (s390x), see:
https://github.com/kleisauke/libvips/commit/673ebdd374e5ebb1ce7ce02198340427da51b906
https://travis-ci.org/github/kleisauke/libvips/jobs/734248676

Although, it fails on this test during `make check`:
https://github.com/libvips/libvips/blob/29b6ba12947a179c650f318354ffd1f57903097d/test/test_formats.sh#L149
```
testing sample.jpg tif[compression=jpeg] ... save / load difference is 263.000000
```

I'm going to try to port that Ruby test case to Python, to see if I can reproduce it on pyvips as well.

**Comment by kleisauke** [comment]
I could also reproduce this on Python with libvips master, see commit https://github.com/kleisauke/libvips/commit/9c2eb79b4e62699b88e89cbb14c0d00e2576c8b3 and [Travis build](https://travis-ci.org/github/kleisauke/libvips/jobs/734262083). Here's a reduced test case: https://github.com/kleisauke/libvips/commit/66b6b2cd8f35c6b6f7dc9cb3b02c43cbf428e494.

**Comment by kleisauke** [comment]
Here's a (possible) patch that seems to work for me: https://github.com/kleisauke/libvips/commit/0690bfea69bd8127cdb527dd324369a5d6e45ce3. The successful Travis build on s390x can be viewed here: https://travis-ci.org/github/kleisauke/libvips/jobs/734595838.

**Comment by kleisauke** [comment]
@jcupitt Do you know if we could safely move the endianness check to compilation time? See for example: https://github.com/kleisauke/libvips/commit/52e77fbe9ad3b4e1c8acb72303a7055b3c2e50e5.

**Comment by kleisauke** [comment]
@jcupitt Let me know if you want a PR with commit https://github.com/kleisauke/libvips/commit/0690bfea69bd8127cdb527dd324369a5d6e45ce3 (commit message needs some polishing) and https://github.com/kleisauke/libvips/commit/52e77fbe9ad3b4e1c8acb72303a7055b3c2e50e5 included. It could possibly target the 8.10 branch.

**Comment by jcupitt** [comment]
Sure, this is a great idea. Thanks for keeping an eye on this, Kleis.

**Comment by LocutusOfBorg** [comment]
thanks for the fix!

**Comment by kleisauke** [comment]
This was fixed in libvips 8.10.3, thanks for reporting this!
